Rewrite the Airflow documentation home page (#24795)
mistercrunch/Airflow: docs/apache-airflow/index.rst

- Workflows can be developed by multiple people simultaneously
- Tests can be written to validate functionality
- Components are extensible and you can build on a wide collection of existing components

Rich scheduling and execution semantics enable you to easily define complex pipelines, running at regular
intervals. Backfilling allows you to (re-)run pipelines on historical data after making changes to your logic.
And the ability to rerun partial pipelines after resolving an error helps maximize efficiency.

Airflow's user interface provides both in-depth views of pipelines and individual tasks, and an overview of
pipelines over time. From the interface, you can inspect logs and manage tasks, for example retrying a task in
case of failure.

The open-source nature of Airflow ensures you work on components developed, tested, and used by many other
`companies <https://github.com/apache/airflow/blob/main/INTHEWILD.md>`_ around the world. In the active
`community <https://airflow.apache.org/community>`_ you can find plenty of helpful resources in the form of
blogs posts, articles, conferences, books, and more. You can connect with other peers via several channels

> such as `Slack <https://s.apache.org/airflow-slack>`_ and a mailing list.

Reviewer: ```suggestion
such as `Slack <https://s.apache.org/airflow-slack>`_ and mailing lists.
```

Nit, we have a couple.

    specific language governing permissions and limitations
    under the License.

What is Airflow?
=========================================

> `Apache Airflow <https://github.com/apache/airflow>`_ is an open-source platform for developing, scheduling,

Reviewer 2: Ilove the summary,  But I think we should mention the "orchestration" here. This is a well know subject. Also I would not be shy - I think we should mention that Airlfow is "As of 2022 when we write this, Airflow is the most popular orchestration platform for batch oriented workflows". I do believe it's truth. I have no hard data to back it up though. 

I think it can have really good "impact" on choosing Airflowl for someone to reach the page.

Reviewer: I think adding orchestration here is probably a good idea, but I say we defer it to another PR as we want to match the wording on the homepage as well.

and monitoring batch-oriented workflows. Airflow's extensible Python framework enables you to build workflows
connecting with virtually any technology. A web interface helps manage the state of your workflows. Airflow is
deployable in many ways, varying from a single process on your laptop to a distributed setup to support even
the biggest workflows.

> Airflow was started in October 2014 by Maxime Beauchemin at Airbnb. It was open source from the very first

Reviewer 2: I think this deserves a separate chapter and it's far less important. While it is cool to be able to look at history, it should be something of "if you are really interested in understanding where it came from go and check it"

Author: Good point. This information is also under https://airflow.apache.org/docs/apache-airflow/stable/project.html so I'll just remove it from here.

of running a Spark job, moving data between two buckets, or sending an email. The same structure can also be
seen running over time:

.. image:: /img/hello_world_grid_view.png
  :alt: Demo DAG in the Grid View, showing the status of all DAG runs

Each column represents one DAG run. These are two of the most used views in Airflow, but there are several
other views which allow you to deep dive into the state of your workflows.

Why Airflow?
=========================================
Airflow is a batch workflow orchestration platform. The Airflow framework contains operators to connect with
many technologies and is easily extensible to connect with a new technology. If your workflows have a clear
start and end, and run at regular intervals, they can be programmed as an Airflow DAG.

If you prefer coding over clicking, Airflow is the tool for you. Workflows are defined as Python code which
means:

- Workflows can be stored in version control so that you can roll back to previous versions
- Workflows can be developed by multiple people simultaneously
- Tests can be written to validate functionality
- Components are extensible and you can build on a wide collection of existing components

> 

Reviewer 2: ```suggestion
- Developing a complex DAG logic or customizing an operator is as easy as using an existing operator. 
  You do not need to learn another language if existing operators are not enough -
  you continue to use Python
```

Reviewer: I see where you are going with these, but I think we can do without.

Rich scheduling and execution semantics enable you to easily define complex pipelines, running at regular
intervals. Backfilling allows you to (re-)run pipelines on historical data after making changes to your logic.
And the ability to rerun partial pipelines after resolving an error helps maximize efficiency.

Airflow's user interface provides both in-depth views of pipelines and individual tasks, and an overview of
pipelines over time. From the interface, you can inspect logs and manage tasks, for example retrying a task in
case of failure.

The open-source nature of Airflow ensures you work on components developed, tested, and used by many other
companies around the world. In the active `community <https://airflow.apache.org/community>`_ you can find
plenty of helpful resources in the form of blogs posts, articles, conferences, books, and more. You can
connect with other peers via several channels such as Slack and a mailing list.

> 

Reviewer 2: ```suggestion
You cam also find out that you are among many companies that use Airflow by exploring `IN_THE_WILD <https://github.com/apache/airflow/blob/main/INTHEWILD.md>` list of companies (and add yours by PR).
```

Author: I'd like to keep this page as concise as possible, IMO the overview should really be a one-pager that only tells the most important information about Airflow. What do you think of linking the word "companies" to INTHEWILD.md?

Reviewer 2: Yeah :). Good point with being concise :)

Why not Airflow?
=========================================
Airflow was built for finite batch workflows. While the CLI and REST API do allow triggering workflows,
Airflow was not built for infinitely-running event-based workflows. Airflow is not a streaming solution.
However, a streaming system such as Apache Kafka is often seen working together with Apache Airflow. Kafka can
be used for ingestion and processing in real-time, event data is written to a storage location, and Airflow
periodically starts a workflow processing a batch of data.

> If you prefer clicking over coding, Airflow is probably not the right solution. The web interface aims to make

Reviewer 2: Love that paragraph :)
